controllers/error-handling.controllers.js

```diff
@@ -1,20 +1,34 @@
+exports.handleCustomErrors = (err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+};
+
 exports.handlePSQL400s = (err, req, res, next) => {
-    if (err.code === "22P02") {
-      res.status(400).send({ msg: "Bad Request" });
-    } else {
-      next(err);
-    }
-  };
-  
-  exports.handleCustomErrors = (err, req, res, next) => {
-    if (err.status && err.msg) {
-      res.status(err.status).send({ msg: err.msg });
-    } else {
-       next(err);
-    }
-  };
-  
-  exports.handle500statuses = (err, req, res, next) => {
-    console.log(err)
-    res.status(500).send({ msg: "Ooopss! Server error!" });
-  };
\ No newline at end of file
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad Request" });
+  } else {
+    next(err);
+  }
+};
+exports.handleForeignKeyErrors = (err, req, res, next) => {
+  if (err.code === "23503") {
+    res.status(400).send({ msg: "something is wrong with your request!" });
```

Comment:
Im guessing you'll just have to change this one to a 404 - username not found to get the test to pass.